src/Gui/Document.cpp

```diff
@@ -556,7 +556,17 @@ ViewProvider * Document::getAnnotationViewProvider(const char* name) const
     return ( (it != d->_ViewProviderMapAnnotation.end()) ? it->second : 0 );
 }
 
-void Document::removeAnnotationViewProvider(const char* name)
+bool Document::isAnnotationViewProvider(const ViewProvider* vp) const
+{
+    std::map<std::string,ViewProvider*>::const_iterator it;
+    for (it = d->_ViewProviderMapAnnotation.begin(); it != d->_ViewProviderMapAnnotation.end(); ++it) {
+        if (it->second == vp)
+            return true;
+    }
+    return false;
+}
+
+void Document::removeAnnotationViewProvider(const char* name, bool deleteVP)

```

Comment:
The second boolean argument is cumbersome because in client code the meaning of true or false is not obvious before checking the function declaration (boolean blindness).

I recommend to remove the second argument and add the new method takeAnnotationViewProvider:
```
ViewProvider* Document::takeAnnotationViewProvider(const char* name)
{
    auto it = d->_ViewProviderMapAnnotation.find(name);
    if (it == d->_ViewProviderMapAnnotation.end()) {
        return nullptr;
    }

    ViewProvider* vp = it->second;
    d->_ViewProviderMapAnnotation.erase(it);

    // cycling to all views of the document
    for (auto vIt : d->baseViews) {
        if (auto activeView = dynamic_cast<View3DInventor *>(vIt)) {
            activeView->getViewer()->removeViewProvider(vp);
        }
    }

    return vp;
}
```
Then removeAnnotationViewProvider() can be re-implemented as
void Document::removeAnnotationViewProvider(const char* name)
{
    delete takeAnnotationViewProvider(name);
}